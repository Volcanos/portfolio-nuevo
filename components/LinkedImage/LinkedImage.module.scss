.imageAnchor {
  display: block;
  &:hover {
    cursor: pointer;
    & img {
      opacity: 0.8;
    }
  }
  & img {
    transition: all 0.2s ease-in-out;
  }
}
