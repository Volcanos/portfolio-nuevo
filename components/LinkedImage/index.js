import Link from 'next/link';
import styles from './LinkedImage.module.scss';

const LinkedImage = ({url, image, title, prefetch = true}) => {
    return (
        <Link href={url} prefetch={prefetch}>
            <a className={styles.imageAnchor}>
                <img src={image} title={title} />
            </a>
        </Link>
    )
}
export default LinkedImage;
