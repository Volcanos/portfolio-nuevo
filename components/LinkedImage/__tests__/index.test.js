import React from 'react'
import { render } from 'enzyme';
import 'jest-styled-components'
import LinkedImage from '../index';

const imageData = {
    url: "/",
    image: "/test.png",
    title: "imagen de prueba"
};

describe('LinkedImage', function () {
    it('should render to title and contain imagen de prueba', function () {
        const text = render(<LinkedImage url={imageData.url} image={imageData.image} title={imageData.title} />).html();
        expect(text).toMatch(/imagen de prueba/);
    });
    it('Basic render snapshot not change', function () {
        const tree = render(<LinkedImage url={imageData.url} image={imageData.image} />);
        expect(tree).toMatchSnapshot();
    });
    it('Complete render snapshot not change', function () {
        const tree = render(<LinkedImage url={imageData.url} image={imageData.image} title={imageData.title} />);
        expect(tree).toMatchSnapshot();
    });
});