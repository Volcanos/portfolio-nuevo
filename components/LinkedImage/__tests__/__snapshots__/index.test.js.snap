// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkedImage Basic render snapshot not change 1`] = `
.c0 {
  display: block;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover img {
  opacity: .8;
}

.c0 img {
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

<a
  class="c0"
>
  <img
    src="/test.png"
  />
</a>
`;

exports[`LinkedImage Complete render snapshot not change 1`] = `
.c0 {
  display: block;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover img {
  opacity: .8;
}

.c0 img {
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

<a
  class="c0"
>
  <img
    src="/test.png"
    title="imagen de prueba"
  />
</a>
`;
